service/source_validation.go

```diff
@@ -55,3 +61,22 @@ func ValidateSourceCreationRequest(dao dao.SourceDao, req *model.SourceCreateReq
 
 	return nil
 }
+
+func ValidateSourceEditRequest(dao dao.SourceDao, editRequest *model.SourceEditRequest) error {
+	if editRequest.Name == nil || *editRequest.Name == "" {
+		return errors.New("name cannot be empty")
+	}
+
+	if dao.NameExistsInCurrentTenant(*editRequest.Name) {
+		return errors.New("name already exists in tenant")
+	}
```

Comment:
I know that this PR is not ready for review but I found out that you solve here the same name validation as was solved here https://github.com/RedHatInsights/sources-api-go/pull/507 (already merged) but the name of function is slightly different so I wanted to let you know
+
for edit request I believe the "Name" field can be empty (when you dont want to edit the name) so the logic should be updated here ... well we need to cover situation when the "name" field is not part of body request and when the "name" field is there and is empty string 🤔

Reply:
In this case I think it is better to just run the validations if the incoming name isn't `nil` — if it is nil, we know it wasn't included in the payload, so we don't need to modify it. If it is not `nil`, even if it was an accidental empty string, we enforce the validation.

Reply:
yes and only for info: this is solved in @Ellen-Yi-Dong #544 too